### PR TITLE
Storybook setup for UI Components

### PR DESCRIPTION
**Pull request by nathanstilwell**
This is a pretty minimal setup on which stories can be build on top of.
It creates a simple placeholder theme, installs a few addons, and has a
simple utility for adding sugar to the knobs addon. All to indicate a
direction that may be used as a pattern for future storybook
installations.

This packages is currently unpublished and no consideration has be given to deployment.

### CODE REVIEW

File: packages/storybook-ui-components/.storybook/cockroach-theme.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { create } from "@storybook/theming/create";
+
+export default create({
+  base: "light",
+
+  // colorPrimary: 'hotpink',
```
[the remark this answers is not in the document]
Comment (nathanstilwell) [reply packages/storybook-ui-components/.storybook/cockroach-theme.js]: I left them there as a placeholder. As we mature our Storybook implementations, I imagine we might want to develop this theme to strengthen the branding. But if they bother you I can remove them. We can always look them up later.

File: packages/storybook-ui-components/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "scripts": {
+    "test": "jest",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cockroachdb/ui.git"
+  },
+  "author": "Nathan Stilwell <nathanstilwell@cockroachlabs.com>",
```
[the remark this answers is not in the document]
Comment (nathanstilwell) [reply packages/storybook-ui-components/package.json]: I'll replace it with the AUTHORS file

File: packages/storybook-ui-components/utils/selectKnobMirror.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const selectKnobMirror = arr =>
```
[the remark this answers is not in the document]
Comment (nathanstilwell) [reply packages/storybook-ui-components/utils/selectKnobMirror.js]: Sorry for this confusion. I'm trying to keep the PRs small and focused, so the usage is in a [different PR](https://github.com/cockroachdb/ui/pull/7/files#diff-dc6e4bc61e2078940e17e50acc521d02R18). This is a utility that helps create shorthand for populating a Select list in a knob. Here's an example,

```javascript
import { select } from "@storybook/addon-knobs";
import mirror from "../utils/selectKnobMirror";

<Component 
  sampleProp={select(
    "Prop Name",
    mirror(["value1", "value2", "value3"]),
   "value1",
  )}
/>
```

compared to

```javascript
import { select } from "@storybook/addon-knobs";

<Component 
  sampleProp={select(
    "Prop Name",
    {
      value1: "value1",
      value2: "value2",
      value3: "value3",
    },
   "value1",
  )}
/>

```

[the remark this answers is not in the document]
Comment (nathanstilwell) [reply packages/storybook-ui-components/utils/selectKnobMirror.js]: 🤦‍♂ I think I just realized that the `select` knob can take an array as options. So please disregard this entire thread.

File: packages/storybook-ui-components/.storybook/cockroach-theme.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (laurenbarker) [reply packages/storybook-ui-components/.storybook/cockroach-theme.js]: Ah gotcha. Yeah I think adding a link to the [theming docs](https://storybook.js.org/docs/configurations/theming/) in a comment and removing the commented out options would make it easier to read.

**Pull request merged**